src/main/java/org/genomenexus/vep_wrapper/VepRunner.java

```diff
@@ -1,125 +1,239 @@
-            stdinWriter.write("\n");
+    /**
+     * Create a file containing the regions received in the input query.
+     * Write the user supplied regions from the "regions" argument to an output file.
+     * CAUTION : this function does not sort the regions into chromosomal order. The
+     * VEP command line tool is very slow when the input is not sorted. It is expected
+     * that users of the VepRunner will always send requests that have been pre-sorted.
+     *
+     * @param regions - the regions as passed by the user
+     * @param vepInputFile - the file to be written
+     * @return sum of two operands
+
+    **/
+    private void constructFileForVepProcessing(List<String> regions, Path vepInputFile) throws IOException {
+        try (PrintWriter out = new PrintWriter(Files.newBufferedWriter(vepInputFile))) {
+            for (String region : regions) {
+                out.println(region);
+            }
+            out.close();
+        } catch (IOException e) {
+            printWithTimestamp("VepRunner : Error - could not construct input file " + vepInputFile);
+            throw e;
         }
-        stdinWriter.flush();
-        stdinWriter.close();
+    }
 
-        //Read output
-        StringBuilder out = new StringBuilder();
-        BufferedReader br = new BufferedReader(new InputStreamReader(process.getInputStream()));
-        String line = null, previous = null;
-        if (convertToListJSON) {
-            out.append('[');
-            out.append('\n');
-        }
-        while ((line = br.readLine()) != null) {
-            if (previous != null && convertToListJSON) {
-                out.append(',');
-                out.append('\n');
+    private Path createTempFileForVepInput() throws IOException {
+        return Files.createTempFile(Paths.get(VEP_TMP_DIRECTORY_PATH), "vep_input-", ".txt");
+    }
+
+    private void forciblyCloseStreamTransfers(StreamTransferrer vepOutputTransferrer, StreamTransferrer vepErrorTransferrer) {
+        printWithTimestamp("requesting shutdown");
+        vepOutputTransferrer.requestShutdown();
+        vepErrorTransferrer.requestShutdown();
+        if (vepOutputTransferrer.isAlive() || vepErrorTransferrer.isAlive()) {
+            printWithTimestamp("waiting 2 seconds..");
+            try {
+                Thread.currentThread().sleep(WAIT_PERIOD_BEFORE_STREAM_CLOSING);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
             }
-            out.append(line);
-            previous = line;
-        }
-        if (convertToListJSON) {
-            out.append(']');
-            out.append('\n');
+            printWithTimestamp("interrupting transferrer threads");
+            vepOutputTransferrer.interrupt();
+            vepErrorTransferrer.interrupt();
+            // TODO: when the destroy of the vep process is made to destory all forked processes as well as the parent process, the code below can be tested/used:
```

Comment:
leftover from dev - cleanup?

Reply:
I think it's actually okay to leave this TODO here. It sort of outlines what we want to accomplish - if it's removed then I suggest creating an issue to address later with this information.